pkg/expression/contextstatic/evalctx.go

```diff
@@ -0,0 +1,362 @@
+func WithLocation(loc *time.Location) StaticEvalCtxOption {
+	intest.AssertNotNil(loc)
+	if loc == nil {
+		// this should not happen, just to keep code safe
+		loc = time.UTC
+	}
+	return func(s *staticEvalCtxState) {
+		s.typeCtx = s.typeCtx.WithLocation(loc)
+	}
+}
+
+// WithErrLevelMap sets the error level map for the `StaticEvalContext`.
+func WithErrLevelMap(level errctx.LevelMap) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.errCtx = s.errCtx.WithErrGroupLevels(level)
+	}
+}
+
+// WithCurrentDB sets the current database name for the `StaticEvalContext`.
+func WithCurrentDB(db string) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.currentDB = db
+	}
+}
+
+// WithCurrentTime sets the current time for the `StaticEvalContext`.
+func WithCurrentTime(fn func() (time.Time, error)) StaticEvalCtxOption {
+	intest.AssertNotNil(fn)
+	return func(s *staticEvalCtxState) {
+		s.currentTime = &timeOnce{timeFn: fn}
+	}
+}
+
+// WithMaxAllowedPacket sets the value of the 'max_allowed_packet' system variable.
+func WithMaxAllowedPacket(size uint64) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.maxAllowedPacket = size
+	}
+}
+
+// WithDefaultWeekFormatMode sets the value of the 'default_week_format' system variable.
+func WithDefaultWeekFormatMode(mode string) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.defaultWeekFormatMode = mode
+	}
+}
+
+// WithDivPrecisionIncrement sets the value of the 'div_precision_increment' system variable.
+func WithDivPrecisionIncrement(inc int) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.divPrecisionIncrement = inc
+	}
+}
+
+// WithPrivCheck sets the requestVerificationFn
+func WithPrivCheck(fn func(db, table, column string, priv mysql.PrivilegeType) bool) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.requestVerificationFn = fn
+	}
+}
+
+// WithDynamicPrivCheck sets the requestDynamicVerificationFn
+func WithDynamicPrivCheck(fn func(privName string, grantable bool) bool) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.requestDynamicVerificationFn = fn
+	}
+}
+
+// WithOptionalProperty sets the optional property providers
+func WithOptionalProperty(providers ...exprctx.OptionalEvalPropProvider) StaticEvalCtxOption {
+	return func(s *staticEvalCtxState) {
+		s.props = contextopt.OptionalEvalPropProviders{}
+		for _, p := range providers {
+			s.props.Add(p)
+		}
+	}
+}
+
+var defaultSQLMode = func() mysql.SQLMode {
+	mode, err := mysql.GetSQLMode(mysql.DefaultSQLMode)
+	if err != nil {
+		panic(err)
+	}
+	return mode
+}()
+
+// StaticEvalContext implements `EvalContext` to provide a static context for expression evaluation.
+type StaticEvalContext struct {
+	id uint64
+	staticEvalCtxState
+}
+
+// NewStaticEvalContext creates a new `StaticEvalContext` with the given options.
+func NewStaticEvalContext(opt ...StaticEvalCtxOption) *StaticEvalContext {
```

Comment:
This and "NewStaticWarnHandlerWithHandler" seems to be no use. Is there any other PR that will use them later?

Reply:
These methods is going to be used in two scenes:

1. replace `mock.Context` to do some tests or evaluate expressions in production code.
2. "Clone" a `SessionEvalContext` for cursor fetch.